mysql/resource_grant.go

```diff
@@ -16,16 +17,231 @@ import (
+	return t.Privileges
+}
+
+func (t *TablePrivilegeGrant) SQLGrantStatement() string {
+	stmtSql := fmt.Sprintf("GRANT %s ON %s.%s TO %s", strings.Join(t.Privileges, ", "), t.GetDatabase(), t.GetTable(), t.UserOrRole.SQLString())
+	if t.TLSOption != "" && strings.ToLower(t.TLSOption) != "none" {
+		stmtSql += fmt.Sprintf(" REQUIRE %s", t.TLSOption)
+	}
+	if t.Grant {
+		stmtSql += " WITH GRANT OPTION"
+	}
+	return stmtSql
+}
+
+func (t *TablePrivilegeGrant) SQLRevokeStatement() string {
+	privs := t.Privileges
+	if t.Grant {
+		privs = append(privs, "GRANT OPTION")
+	}
+	return fmt.Sprintf("REVOKE %s ON %s.%s FROM %s", strings.Join(privs, ", "), t.GetDatabase(), t.GetTable(), t.UserOrRole.SQLString())
+}
+
+func (t *TablePrivilegeGrant) SQLPartialRevokePrivilegesStatement(privilegesToRevoke []string) string {
+	if t.Grant {
+		privilegesToRevoke = append(privilegesToRevoke, "GRANT OPTION")
+	}
+	stmt := fmt.Sprintf("REVOKE %s ON %s.%s FROM %s", strings.Join(privilegesToRevoke, ", "), t.GetDatabase(), t.GetTable(), t.UserOrRole.SQLString())
+	return stmt
+}
+
+type ProcedurePrivilegeGrant struct {
+	Database     string
+	ObjectT      ObjectT
+	CallableName string
+	Privileges   []string
+	Grant        bool
+	UserOrRole   UserOrRole
+	TLSOption    string
+}
+
+func (t *ProcedurePrivilegeGrant) GetId() string {
+	return fmt.Sprintf("%s:%s", t.UserOrRole.IDString(), t.GetDatabase())
+}
+
+func (t *ProcedurePrivilegeGrant) GetUserOrRole() UserOrRole {
+	return t.UserOrRole
+}
+
+func (t *ProcedurePrivilegeGrant) GrantOption() bool {
+	return t.Grant
+}
+
+func (t *ProcedurePrivilegeGrant) GetDatabase() string {
+	if strings.Compare(t.Database, "*") != 0 && !strings.HasSuffix(t.Database, "`") {
+		return fmt.Sprintf("`%s`", t.Database)
+	}
+	return t.Database
+}
+
+func (t *ProcedurePrivilegeGrant) GetPrivileges() []string {
+	return t.Privileges
+}
+
+func (t *ProcedurePrivilegeGrant) SQLGrantStatement() string {
+	stmtSql := fmt.Sprintf("GRANT %s ON %s %s.%s TO %s", strings.Join(t.Privileges, ", "), t.ObjectT, t.GetDatabase(), t.CallableName, t.UserOrRole.SQLString())
+	if t.TLSOption != "" && strings.ToLower(t.TLSOption) != "none" {
+		stmtSql += fmt.Sprintf(" REQUIRE %s", t.TLSOption)
+	}
+	if t.Grant {
+		stmtSql += " WITH GRANT OPTION"
+	}
+	return stmtSql
+}
+
+func (t *ProcedurePrivilegeGrant) SQLRevokeStatement() string {
+	privs := t.Privileges
+	if t.Grant {
+		privs = append(privs, "GRANT OPTION")
+	}
+	stmt := fmt.Sprintf("REVOKE %s ON %s %s.%s FROM %s", strings.Join(privs, ", "), t.ObjectT, t.GetDatabase(), t.CallableName, t.UserOrRole.SQLString())
+	return stmt
+}
+
+func (t *ProcedurePrivilegeGrant) SQLPartialRevokePrivilegesStatement(privilegesToRevoke []string) string {
+	stmt := fmt.Sprintf("REVOKE %s ON %s %s.%s FROM %s", strings.Join(privilegesToRevoke, ", "), t.ObjectT, t.GetDatabase(), t.CallableName, t.UserOrRole.SQLString())
+	if t.Grant {
+		stmt += " WITH GRANT OPTION"
+	}
```

Comment:
Sorry, I just noticed this remained in place (I did not comment on both of these).